ui/packages/shared/profile/src/contexts/MatchersInputLabelsContext.tsx

```diff
@@ -54,7 +54,7 @@ export function LabelsProvider({
   end,
 }: LabelsProviderProps): JSX.Element {
   const [currentLabelName, setCurrentLabelName] = React.useState<string | null>(null);
-  const utilizationLabels = useUtilizationLabels();
+  const utilizationContext = useUtilization();
```

Comment:
Maybe we should try an approach where we don't need to deal with both the labels and determine which to use here. 
IMO if there is a way we can make this generic to take only the labelNames and fetchLabelsNames handler  and not necessary to know the type of labels that would be great.

Reply:
i think i have an idea for this🤔 i'll try it out and let you know